samples/bluetooth/direction_finding_connectionless_tx/src/main.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <sys/printk.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/direction.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+
+/* Length of CTE in unit of 8[us] */
+#define CTE_LEN (0x14U)
+
+static void adv_sent_cb(struct bt_le_ext_adv *adv,
+			struct bt_le_ext_adv_sent_info *info);
+
+static struct bt_le_ext_adv_cb adv_callbacks = {
+	.sent = adv_sent_cb,
+};
+
+static struct bt_le_ext_adv *adv_set;
+
+static struct bt_le_adv_param param =
+		BT_LE_ADV_PARAM_INIT(BT_LE_ADV_OPT_EXT_ADV |
+				     BT_LE_ADV_OPT_USE_NAME,
+				     BT_GAP_ADV_FAST_INT_MIN_2,
+				     BT_GAP_ADV_FAST_INT_MAX_2,
+				     NULL);
+
+static struct bt_le_ext_adv_start_param ext_adv_start_param = {
+	.timeout = 0,
+	.num_events = 0,
+};
+
+static struct bt_le_per_adv_param per_adv_param = {
+	.interval_min = BT_GAP_ADV_SLOW_INT_MIN,
+	.interval_max = BT_GAP_ADV_SLOW_INT_MAX,
+	.options = BT_LE_ADV_OPT_USE_TX_POWER,
+};
+
+#if defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_TX)
+/* Example sequence of antenna switch patterns for antenna matrix designed by
+ * Nordic. For more information about antenna switch patterns see README.rst.
+ */
+static uint8_t ant_patterns[] = {0x2, 0x0, 0x5, 0x6, 0x1, 0x4, 0xC, 0x9, 0xE,
```

Comment:
const?

Reply:
maybe with above structures too?

Reply:
I've added const where it was possible (where it don't require casting due to non-const pointers e.g. `.ant_ids = ant_patterns`).
I'll update DF interface in Zephyr to use const pointers were it is possible and then get back to sample to update it as well.